### PR TITLE
Remove Wildcard's dependency on @sourcegraph/common

### PR DESCRIPTION
**Pull request by vdavid**
This is part of the effort to publish Wildcard on NPM.

I heard from @vovakulikov that when he had looked into the possibility of publishing Wildcard 1–2 months ago, his only concern was the large bundle size because Wildcard depends on `@sourcegraph/common`. He listed these two options to solve this:
a.) prebuild common and publish it as well (we wouldn’t do it publish cascade is not easy to maintain)
b.) build wildcard only and publish it with the hanging dependency (it adds some complexity to the consumer)

I wasn't fond of these and wanted to find a third option. So I looked into how much _actual_ dependency there was from Wildcard to `@sourcegraph/common`, and it turned out that there wasn't too much. Only 11 files, 15 specific imports. I looked into each of the 15. Here is the list of them, with some comments on how complex they are:
- Trivial one-liners
    - isDefined
    - anyOf
    - isInstanceOf
    - type HistoryOrNavigate (but depends on React Router, not sure if that’s OK)
- Fairly simple ones
    - logger (a simple interface and very simple implementation)
    - asError
    - createAggregateError
    - isExternalLink
    - compatNavigate
    - getBrowserName
- Might be tricky
    - ALL_LANGUAGES (only used in a test but long, and we probably don’t want to duplicate this)
    - highlightCodeSafe (simple, but depends on highlight.js)
    - registerHighlightContributions (not too elaborate, but depends on highlight.js)
    - renderMarkdown (150 code lines, depends on `marked` and `dompurify`)

So here is what I did with these:
- The trivial and "fairly simple" ones I just copied to Wildcard. Yes, it's some code duplication, but it seems manageable. These are utility functions we don't want to change around often.
- ALL_LANGUAGES: I dropped the test. It seemed like a worthwhile sacrifice, and better than duplicating this large constant that might get stale quickly or keeping the Wildcard→common reference.
- `highlightCodeSafe`, `registerHighlightContributions`, and `renderMarkdown`: Copy-pasted this code as well, but marked it with a TODO to somehow resolve this later. Maybe publishing our markdown-related functionality in a third, smaller package could be viable as it's not likely to change often.

The next step after getting this approved is to automate publishing Wildcard from CI. For that, I can work with someone from dev-infra team.

## Test plan

I've ran `sg` locally and clicked around to see if Wildcard loads and generally works smoothly (it does), but I haven't checked the specific points where the modified components display.
I'll do that before merging the PR.

### CODE REVIEW

**Review by vovakulikov — APPROVED** [review]
Left some comments about missing deps packages. 

One thought on publishing: I suppose we don't want to compile anything (TS, SCSS files) before publishing because it would lead to a problem when scss styles, which means that we wouldn't be able to import and use mixins and scss helpers from wildcard in the new client, it also causes some problem with css duplications. 

If we publish just codebase with no compile step we need to check that client bundlers which is going to use wildcard package handles styles and ts files from node_modules properly. 

Approve to unblock. If there are going to be some follow-up problems, we can resolve them as they appear. Thanks for the effort.

File: client/wildcard/src/utils/markdown.ts [hunk 1]
```diff
@@ -0,0 +1,150 @@
+import DOMPurify, { type Config as DOMPurifyConfig } from 'dompurify'
```
[the remark this answers is not in the document]
Comment (vdavid) [reply client/wildcard/src/utils/markdown.ts:1]: Yeah, I was wondering the same about `react-router-dom` and `dompurify`. I'll add them as dependencies, along with `marked`, which, I think, is a third one.

File: client/wildcard/package.json [hunk 2]
```diff
@@ -13,9 +13,10 @@
     "test": "vitest"
   },
   "devDependencies": {
-    "@sourcegraph/testing": "workspace:*"
+    "@sourcegraph/testing": "workspace:*",
+    "@types/highlight.js": "^9.12.4"
   },
   "dependencies": {
-    "@sourcegraph/common": "workspace:*"
+    "highlight.js": "^10.7.3"
```
[the remark this answers is not in the document]
Comment (vdavid) [reply client/wildcard/package.json:20]: Sorry, could you rephrase? What would you check exactly?
By "blocks UI", do you mean you're worried about whether things _function_ at all, or more like about _performance_? (If performance, then what's the diff? I mean, we're still loading the same libraries as before, right?)

**Comment by vdavid** [comment]
@vovakulikov 

> One thought on publishing: I suppose we don't want to compile anything (TS, SCSS files) before publishing because it would lead to a problem when scss styles, which means that we wouldn't be able to import and use mixins and scss helpers from wildcard in the new client, it also causes some problem with css duplications.
> If we publish just codebase with no compile step we need to check that client bundlers which is going to use wildcard package handles styles and ts files from node_modules properly.

I'll rephrase this to make sure I understand your comment.
What I'm hearing is that you're arguing against compiling TS and SCSS before publishing the package because you believe it'd cause some issues. Instead, you're suggesting to publish the codebase as-is, but if we do that, we need to make sure that we include the instructions to build the TS and SCSS in the README.md of the package, so that clients (currently PLG, but hopefully will be others) will set up the necessary tooling in the right way for Wildcard to function out in the _wild_ (🥁).

Is my understanding correct?

**Review by eseliger — COMMENTED** [review]
Thanks for taking the time to finally untangle this!

To fully codify this change, I think we should remove common from here as well: https://sourcegraph.com/-/editor?remote_url=github.com%2Fsourcegraph%2Fsourcegraph&branch=main&file=client%2Fwildcard%2Ftsconfig.json&editor=VSCode&version=2.2.16&start_row=9&start_col=39&end_row=9&end_col=39

I'll defer to Vova for final approval, as he has a much better understanding of this package :)

File: client/wildcard/package.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (vovakulikov) [reply client/wildcard/package.json]: I worry that maybe some global CSS classes have been changed in the 10 version of highlight js, I would visually check our code highlighted blocks (like the ones on the search result page). Performance should be fine

**Comment by vovakulikov** [comment]
@vdavid 
> Is my understanding correct?

this is my point exactly, yes.

File: client/wildcard/package.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (vdavid) [reply client/wildcard/package.json]: Oh, I see! I didn't realize that I updated the version.
I can see now that in the repo root, we use `"highlight.js": "^10.5.0"` and `"@types/highlight.js": "^9.12.4"`. (The reason for the diff is that there's no newer type package available.), while I used `"highlight.js": "^10.7.3"` in Widcard. This was unintentional, so I'll just revert it to `"highlight.js": "^10.5.0"` to avoid this PR being thicker than needed.

[the remark this answers is not in the document]
Comment (vdavid) [reply client/wildcard/package.json]: Reverted it in https://github.com/sourcegraph/sourcegraph/pull/61781/commits/bc321d59d7d31463691d2eb60bd8f86b715fbe94
